sources/packages/api/src/route-controllers/restriction/restriction.aest.controller.ts

```diff
@@ -164,14 +166,16 @@ export class RestrictionController extends BaseController {
    * @param studentId
    * @param payload
    */
-  @Groups(UserGroups.AESTUser)
-  @AllowAuthorizedParty(AuthorizedParties.aest)
   @Roles(Role.StudentAddRestriction)
+  @ApiUnprocessableEntityResponse({
+    description:
+      "The given restriction type is not Provincial. Only provincial restrictions can be added.",
+  })
   @Post("/student/:studentId")
   async addStudentProvincialRestriction(
```

Comment:
This method is not using a transaction to save the restriction and the student note, can you please add a todo to refactor this?
![image](https://user-images.githubusercontent.com/61259237/184968219-f27a078c-3335-4d06-8697-7ffb5d089395.png)